File: recipes/jvm/java-recipes/quarkus-basic-recipe/src/main/java/com/trendyol/stove/recipes/quarkus/StoveStartupSignal.java
package com.trendyol.stove.recipes.quarkus;

import io.quarkus.runtime.ShutdownEvent;
import io.quarkus.runtime.StartupEvent;
import jakarta.enterprise.context.ApplicationScoped;
import jakarta.enterprise.event.Observes;

@ApplicationScoped
public class StoveStartupSignal {

  public static final String READY_PROPERTY = "stove.quarkus.ready";

  void onStart(@Observes StartupEvent event) {
    System.setProperty(READY_PROPERTY, "true");
  }

  void onStop(@Observes ShutdownEvent event) {
    System.clearProperty(READY_PROPERTY);
  }
}

File: examples/quarkus-example/src/main/kotlin/stove/quarkus/example/StoveStartupSignal.kt
package stove.quarkus.example

import io.quarkus.runtime.ShutdownEvent
import io.quarkus.runtime.StartupEvent
import jakarta.enterprise.context.ApplicationScoped
import jakarta.enterprise.event.Observes

@Suppress("unused")
@ApplicationScoped
class StoveStartupSignal {
  fun onStart(
    @Observes event: StartupEvent
  ) {
    System.setProperty(READY_PROPERTY, "true")
  }

  fun onStop(
    @Observes event: ShutdownEvent
  ) {
    System.clearProperty(READY_PROPERTY)
  }

  companion object {
    const val READY_PROPERTY: String = "stove.quarkus.ready"
  }
}
